lib/new_relic/agent/instrumentation/action_mailbox.rb

```diff
@@ -0,0 +1,26 @@
+# This file is distributed under New Relic's license terms.
+# See https://github.com/newrelic/newrelic-ruby-agent/blob/main/LICENSE for complete details.
+# frozen_string_literal: true
+
+require 'new_relic/agent/instrumentation/action_mailbox_subscriber'
+
+DependencyDetection.defer do
+  named :action_mailbox
+
+  depends_on do
+    defined?(ActiveSupport) &&
+      defined?(ActionMailbox) &&
+      ActionMailbox.respond_to?(:gem_version) && # 'require "action_mailbox"' doesn't require version...
+      ActionMailbox.gem_version >= Gem::Version.new('7.1.0.alpha') && # notifications added in Rails 7.1
+      !NewRelic::Agent::Instrumentation::ActionMailboxSubscriber.subscribed?
+  end
+
+  executes do
+    NewRelic::Agent.logger.info('Installing ActionMailbox instrumentation')
+  end
+
+  executes do
+    ActiveSupport::Notifications.subscribe(/^process\.action_mailbox$/,
```

Comment:
What do you think about using a wildcard match instead of `process`? I know it's different than the original ticket, but it would be a bit more future-proof.

Reply:
Also, could you add a link to the code where the events are defined for ActionMailbox? I'd like to see them/understand the keys in the payload.

Reply:
My take would be to hold off on wildcarding for now, and when we circle back after having some minimal coverage for every event type we can discuss perhaps standardizing on a wildcard approach for all subscribers. Given that this ActionMailbox class only has a single event that's still in alpha, I'm also leery of updating it in particular for future proofing, but I can certainly see how `./^*\.action_mailbox$/` would work. Maybe on the circle back we can just do `./^.*_#{subscriber_class}$/` dynamically or whatnot.

Reply:
The Rails Edge code that delivers the lone ActionMailbox event is here: https://github.com/rails/rails/blob/246c13f8df833cd8d8ae423af253bf06ceca0fc0/actionmailbox/lib/action_mailbox/base.rb#L81

Reply:
Thank you! Let's hold off then.

Reply:
Switched to a wildcard with [354f0de](https://github.com/newrelic/newrelic-ruby-agent/pull/1740/commits/354f0de483791f52daec5bae941b6e513015e127)